fog/sample-paykit/src/cached_tx_data/memo_handler.rs

```diff
@@ -149,6 +149,13 @@ impl MemoHandler {
                     Err(MemoHandlerError::FailedSubaddressValidation)
                 }
             }
+            MemoType::Defragmentation(_) => {
+                if subaddress_matches_tx_out(account_key, CHANGE_SUBADDRESS_INDEX, tx_out)? {
+                    Ok(Some(memo_type))
+                } else {
+                    Err(MemoHandlerError::FailedSubaddressValidation)
+                }
+            }
```

Comment:
The second option seems more reasonable to me but I don't have a strong preference either way, it might be a nice thing to correct later but I'd be fine with the PR getting merged with this code as-is.